wallet-new/src/Cardano/Wallet/Kernel.hs

```diff
@@ -300,24 +304,44 @@ cancelPending passiveWallet cancelled =
     update' (passiveWallet ^. wallets) $ CancelPending (fmap InDb cancelled)
 
 {-------------------------------------------------------------------------------
-  Wallet Account read-only API
+                              Wallet getters
+
+  The @only@ effectful function we expose is 'getWalletSnapshot', which reads
+  the full DB 'Snapshot' and returns it.
+  All the other getters are completely pure and take the 'Snapshot' as input,
+  so that users of the wallet are forced to re-use the same 'Snapshot' in case
+  they want to read the state of the wallet multiple times within the same
+  code block / handler.
+
 -------------------------------------------------------------------------------}
 
+-- | A type synonym to stress the fact that what we are carrying around is not
```

Comment:
Which type synonym does this refer to?

Reply:
Good catch! I've previously introduced a type synonym `type DbSnapshot = DB` but decided later on it was not worth the cognitive overload. Will remove the stale comment!